### PR TITLE
fix(preset-icons): support icon name contains `_`

### PR DESCRIPTION
**Pull request by mokeyish**
Icon names that contain underscores cannot be recognized. such as `i-xx-a_b`.

### CODE REVIEW

**Review by antfu — APPROVED** [review]

**Review by antfu — APPROVED** [review]

File: packages/preset-icons/src/core.ts [hunk 1]
```diff
@@ -57,7 +57,7 @@ export function createPresetIcons(lookupIconLoader: (options: IconsOptions) => P
       options,
       layers: { icons: -30 },
       rules: [[
-        /^([a-z0-9:-]+)(?:\?(mask|bg|auto))?$/,
+        /^([a-z0-9:-_]+)(?:\?(mask|bg|auto))?$/,
```
Comment (userquin) [inline packages/preset-icons/src/core.ts:60]: set `-` last or escape it

Comment (userquin) [reply packages/preset-icons/src/core.ts:60]: `/^([a-z0-9:_-]+)(?:\?(mask|bg|auto))?$/`

**Comment by mokeyish** [comment]
The latest version `i-gitlab-svgs-status_running` displays black circle. How to fix this?

图片


```html
          <i class="i-gitlab-svgs-template"></i>
          <i class="i-gitlab-svgs-status_running"></i>
```
- status_running 
- status_notfound 

The configuration:
```js
presetIcons({
        cdn: 'https://esm.sh/',
        warn: true,
        extraProperties: {
          display: 'inline-block'
        },
        collections: {
          'gitlab-svgs': async (name: string) => {
            // preview https://gitlab-org.gitlab.io/gitlab-svgs/
            return $fetch(`https://cdn.jsdelivr.net/gh/mokeyish/gitlab-svgs-o@v3.3.0/icons/sprite_icons/${name}.svg`, { responseType: 'text' });
          }
        }
      })

```

**Comment by mokeyish** [comment]
It seems conflict with `status_`.

图片

**Comment by mokeyish** [comment]
@antfu @userquin Hi, do you have any ideas to solve the issue above?

**Comment by userquin** [comment]
@mokeyish try render it without color: `i-gitlab-svgs-status_running?bg`; I guess the svg is wrong, using `fill="currentColor"`

[playground](https://unocss.dev/play/?html=DwEwlgbgBAFgtAMwK4BsVQC4FMAeG4DGWAdtgE5QIq5QDOW1B%2BxA9sVlAIZoBcGZnYrTAYwbOABYADFIB8AKChRQkKAFtOCpUpXRseOAFYc6BAHcAjDK7EwG7HABGLJMSJxuGG3c4OCL0jgwYgRgkQ5BHwcQJAFRcQtaLW0lAFVWAGEAZSzFbWAAenAIZJ1iqBYABwBmKUxcfBQAc0ozWrq1C1LtABUYDmDaDEEvcRAsDWIQKABBDBY1MAIobKyoEibgrAA6PJ0iyG7ddQAmSmocKAArJCGwBABPQhJyeoMTkwqaupgWCCwyABeABEVQAHFJgd0dJw9ikoGBCJwyM5iHAUCwmiw4JsMDAkI44Sl9Phgv0yCIidoYGQsAgQTAMBhKrQeAUCrj8Y5tv41AVXCwCLRaPzWELaMCqUphmQmlgMCCAPqOFCCADWkvhUFkhU0VOOBFVwpBiJwPAeAH5HE1gVACkcDiU8oVigoXYd5MdOI5aCwUEhsFBnEyFkYoBSmoy4HVqAh8HUSc9SACvrVWu1kmMJoJpgB2hFCYakKD55BoB5QAi3eaLABe3uonsdsiAA&config=JYWwDg9gTgLgBAbwFBzgEwKYDNgDsMDCEuOA5gDQpxhQYDOGMAgjDFMAEYCuMwWAnpVQ16jAJIBjYnSHVaDGAFVcESgF84WKBBBwA5FxUS6dPUiQYAHpFjpsAQy4AbeJhz4iJYKQAUyVFDO9ABccADaVKhhehJcdDA6ALSBThh65IhwUk7QoXq0aHpwagC6VCWydAAW0DCxMHSh-qj6sfFJ1bX1enkwVjCJTqRwfZYD0Pa4pBhwNQBuGFDBowN99k5mqGqyIgqN4ZFyokoqPgCUssLyjCxsnDx8-OeXRwqS0n6HqHQS6xihAEYAHQAJheqAkaFweSqrDAjQA9Aj6CAgdUEekvlkIE5UhJeNImljUJYiS1yS1%2BKEAAYAHjoc2GlhATlwdAAvAAiWEwMDBJEAdyFQIFAGYgdBSAiQQAGOUIhmkTlwAXANAwKpcgEAFmVVQw3lhWt1cDmwAwAoAQhBLFyZXB7Tq4DrlThcVzYlBaLgYEQclBOQA%2BKi04ZupzJIJcjALFRoNBBkNgewa9BcgCy9oA7PYs3A847nc7tQ6mHmCw6i9mAF6chHB1C05OptAZgGi-NMABscB7hcLOZ7fcrjpBDtrmmA7s5AGIAGIL5XxfipLnh4JzexQHyJRKKvcwFNxRIcChwGdYS9nZX1pMpqppznpvOiiQg0GVkvAgAccBL2roRIfz-RJ-wkICgQ7GVEnfABWRIgSzRJxVFCCswAGXLAA1UU6wbOBaQRUhg0IxVA2pYliixNRDjUC5yiQOikCAA&css=Q&options=N4XyA)

**Pull request merged**